ee/codegen/src/generators/nodes/bases/base.ts

```diff
@@ -779,12 +779,22 @@ export abstract class BaseNode<
       );
     }
 
-    nodeClass.add(
-      python.field({
-        name: "node_id",
-        initializer: python.TypeInstantiation.uuid(this.nodeData.id),
-      })
-    );
+    // Only add node_id if it differs from the hash-generated UUID
+    // Use the node's definition module path (matching Python SDK's node_class.__module__)
+    const definitionModule = this.nodeData.definition?.module;
+    const definitionName = this.nodeData.definition?.name;
+    const expectedNodeId =
+      definitionModule && definitionName
+        ? getNodeIdFromModuleAndName(definitionModule, definitionName)
+        : undefined;
+    if (expectedNodeId === undefined || this.nodeData.id !== expectedNodeId) {
```

Comment:
DevinAI comment out this check for now - there are some vembda-side problems we need to solve first